src/ShowingTransitionSupport.js

```diff
@@ -0,0 +1,212 @@
+* @public
+*/
+module.exports = {
+
+	/**
+	* @private
+	*/
+	name: 'ShowingTransitionSupport',
+
+	/**
+	* A read-only property for checking whether we are in the middle of a transition.
+	*
+	* @type {Boolean}
+	* @default false
+	* @public
+	*/
+	showingTransitioning: false,
+
+	/**
+	* The amount of time the "showing" transition takes to complete in milliseconds.
+	*
+	* @type {Number}
+	* @default undefined
+	* @public
+	*/
+	showingDuration: undefined,
+
+	/**
+	* The amount of time the "hiding" transition takes to complete in milliseconds.
+	*
+	* @type {Number}
+	* @default undefined
+	* @public
+	*/
+	hidingDuration: undefined,
+
+	/**
+	* The method to fire at the end of the "showing" transition. This may be name of a method on the
+	* component, or a function.
+	*
+	* @type {String|Function}
+	* @default undefined
+	* @public
+	*/
+	shownMethod: undefined,
+
+	/**
+	* The method to fire at the end of the "hiding" transition. This may be name of a method on the
+	* component, or a function.
+	*
+	* @type {String|Function}
+	* @default undefined
+	* @public
+	*/
+	hiddenMethod: undefined,
+
+	/**
+	* The the classname to apply for the "shown" (component is visible) resting state.
+	*
+	* @type {String}
+	* @default undefined
+	* @public
+	*/
+	shownClass: undefined,
+
+	/**
+	* The the classname to apply for the "hidden" (component is not visible) resting state.
+	*
+	* @type {String}
+	* @default 'hidden'
+	* @public
+	*/
+	hiddenClass: undefined,
+
+	/**
+	* The the classname to apply for the "hiding" (component has started the transition to the
+	* hidden state) transition state.
+	*
+	* @type {String}
+	* @default 'hiding'
+	* @public
+	*/
+	hidingClass: undefined,
+
+	/**
+	* The the classname to apply for the "showing" (component has started the transition to the
+	* shown state) transition state.
+	*
+	* @type {String}
+	* @default 'showing'
+	* @public
+	*/
+	showingClass: undefined,
+
+	/**
+	* Initializes the defaults, and prepares the component with its classes in case of initially
+	* "showing: false".
+	*
+	* @method
+	* @private
+	*/
+	create: kind.inherit(function (sup) {
+		return function () {
+			sup.apply(this, arguments);
+			this.showingDuration = (this.showingDuration === undefined) ? null      : this.showingDuration;
+			this.hidingDuration  = (this.hidingDuration  === undefined) ? null      : this.hidingDuration;
+			this.shownMethod     = (this.shownMethod     === undefined) ? null      : this.shownMethod;
+			this.hiddenMethod    = (this.hiddenMethod    === undefined) ? null      : this.hiddenMethod;
+			this.shownClass      = (this.shownClass      === undefined) ? null      : this.shownClass;
+			this.hiddenClass     = (this.hiddenClass     === undefined) ? 'hidden'  : this.hiddenClass;
+			this.hidingClass     = (this.hidingClass     === undefined) ? 'hiding'  : this.hidingClass;
+			this.showingClass    = (this.showingClass    === undefined) ? 'showing' : this.showingClass;
+			this.showingChanged();
```

Comment:
My hesitance to merge right away is not based on any specific concerns with this idea or its implementation, but on a) Not yet having had sufficient time to thoroughly experiment with this feature at the framework or application level; and b) The sense that we will want to make other view-management-lifecycle changes in the relatively near future and a desire to think about them holistically before we commit to any new API.

Also, I want to get 2.7 out the door as quickly as possible and with as little risk as possible so that we can start doing some comparatively bold and experimental stuff on master without worrying about near-term releases . With that goal in mind, the bar for what gets into 2.7 is very high -- basically, things go in only if they absolutely must.

All that being said, I would be OK with this going in as WIP, as long as we don't think there's any risk to existing behavior -- that is, if there is no impact on apps unless / until they start using the feature.